File: src/main/java/no/nav/sbl/service/unleash/UnleashServiceImpl.java
package no.nav.sbl.service.unleash;

import io.getunleash.Unleash;

public class UnleashServiceImpl implements UnleashService {
    private final Unleash defaultUnleash;

    public UnleashServiceImpl(Unleash defaultUnleash) {
        this.defaultUnleash = defaultUnleash;
    }

    @Override
    public boolean isEnabled(Feature feature) {
        return defaultUnleash.isEnabled(feature.getPropertyKey());
    }

    @Override
    public boolean isEnabled(String feature) {
        return defaultUnleash.isEnabled(feature);
    }
}

File: src/main/java/no/nav/sbl/service/unleash/UnleashServiceImpl.kt
package no.nav.sbl.service.unleash

import io.getunleash.Unleash

class UnleashServiceImpl(
    private val defaultUnleash: Unleash
) : UnleashService {
    override fun isEnabled(feature: Feature): Boolean {
        return defaultUnleash.isEnabled(feature.propertyKey)
    }

    override fun isEnabled(feature: String): Boolean {
        return defaultUnleash.isEnabled(feature)
    }
}
